JASP-Engine/JASP/R/reliabilityanalysis.R

```diff
@@ -15,479 +15,456 @@
-		fields[[length(fields) + 1]] <- list(name="lower", title="Lower", type="number", format="sf:4;dp:3", overTitle = overTitle)
-		fields[[length(fields) + 1]] <- list(name="upper", title="Upper", type="number", format="sf:4;dp:3", overTitle = overTitle)
-	}
-
-	table[["schema"]] <- list(fields = fields)
-
-	data <- list()
-
-	if (!is.null(r)) {
-
-		footnotes <- .newFootnotes()
-
-		if (options[["missingValues"]] == "excludeCasesListwise") {
-
-			exclwise = " listwise"
-
-		} else {
-
-			exclwise = " pairwise"
-
-		}
-
-		nObs <- nrow(dataset)
-		nExcluded <- sum(!complete.cases(dataset))
-		nValid <- nObs - nExcluded
-
-		# message <- paste("Scale consists of items ", paste0(variables, collapse = ", "))
-		message <- sprintf("Of the observations, %d were used, %d were excluded%s, and %d were provided.",
-						   nValid, nExcluded, exclwise, nObs)
-
-		if (options[["glbScale"]]) {
-
-			if (length(variables) <= 2) {
-
-				message <- paste(message, "Warning: Greatest lower bound can only be calculated for three or more variables.")
-
-			} else if (isTryError(r[["glb"]])) {
-
-				message <- paste(message, "Warning: Greatest lower bound could not be calculated.")
-
-			}
-
-		}
-
-		.addFootnote(footnotes, symbol = "<em>Note.</em>", text = message)
-
-		table[["footnotes"]] <- as.list(footnotes)
-
-
-		alpha <- NULL
-		lambda <- NULL
-		mu <- NULL
-		sd <- NULL
-		rho <- NULL
-		omega <- NULL
-		glb <- NULL
-		lower <- NULL
-		upper <- NULL
-
-		if (options[["alphaScale"]])
-            alpha <- .clean(r$total[[options$alphaNms]])
-
-		if (options$gutmannScale)
-			lambda <- .clean(r$total[["G6(smc)"]])
-
-		if (options$meanScale)
-			mu <- .clean(r$total$mean)
-
-		if (options$sdScale)
-			sd <- .clean(r$total$sd)
-
-		if (options[["averageInterItemCor"]])
-			rho <- .clean(r[["total"]][["average_r"]])
-
-		if (options[["mcDonaldScale"]])
-			omega <- .clean(r[["omega"]])
-
-		if (options[["glbScale"]]) {
-			if (r[["glb"]] == "." || isTryError(r[["glb"]])) { # unusable information
-				glb <- "."
-			} else { # a useable value
-				glb <- .clean(r[["glb"]])
-			}
-		}
-
-		if (options[["confAlpha"]]) {
-			lower = .clean(r[["ciAlpha"]][1])
-			upper = .clean(r[["ciAlpha"]][2])
-		}
-
-		data[[1]] <- list(case="scale", alpha=alpha, lambda=lambda, omega = omega, glb = glb, rho=rho, mu=mu, sd=sd, lower = lower, upper = upper)
-
-		table[["status"]] <- "complete"
-
-	} else {
-
-		data[[1]] <- list(case="scale", alpha=".", lambda=".", omega = ".", glb = ".", rho =".", mean=".", sd=".", lower = ".", upper = ".")
-
-	}
-
-	table[["data"]] <- data
-
-	return(table)
-
+.reliabilityScaleFill <- function(jaspResults, dataset, options, relyFit, ready) {
+  results   <- list()
+  nObs      <- nrow(dataset)
+  nVar      <- ncol(dataset)
+  variables <- unlist(options$variables)
+  options   <- .reliabilitySetAlphaNms(options)
+  
+  if(ready){
+    #Initialise all as null
+    alpha <- lambda <- omega <- glb <- rho <- mu <- sd <- lower <- upper <- NULL
```

Comment:
You don't have to pre-initialize these. In your jaspTable you can set `$showSpecifiedColumnsOnly <- TRUE` so only the columns which have their info set through `$addColumnInfo()` will be shown. All the other columns -- even those with data -- are discarded. This also removes the need for all the if statements below.